include/domain/impl/receivers/kernel.tpp

```diff
@@ -186,10 +194,110 @@ void specfem::domain::impl::kernels::receiver_kernel<
                                  point_properties,
                                  element_quadrature.hprime_gll, active_field,
                                  sv_receiver_field);
+              if(seismogram_type_l == specfem::enums::seismogram::type::pressure){
+                //for pressure, we need to compute -kappa * div(s)
+                //so we store displacement into auxfield so we can overwrite
+                //sv_receiver_field
+
+                #ifndef KOKKOS_ENABLE_CUDA
+                #pragma unroll
+                #endif
+                for (int l = 0; l < specfem::dimension::dimension<DimensionType>::dim; l++) {
+                  aux_field.displacement(iz,ix,l) = sv_receiver_field(l);
```

Comment:
With direct access to Kokkos::View as stated in the comment above. This could be `aux_field(iz, ix, l)`